File: DyeHard/Java/DyeHardAPI/src/dyehard/Collectibles/DyePack.java
package dyehard.Collectibles;

import java.awt.Color;

import Engine.BaseCode;
import Engine.Vector2;
import dyehard.Collidable;
import dyehard.Configuration;
import dyehard.Player.Hero;
import dyehard.Util.Colors;
import dyehard.Util.DyeHardSound;
import dyehard.Util.Timer;

public class DyePack extends Collidable {
    public static final float width = Configuration.dyePackWidth;
    public static final float height = Configuration.dyePackHeight;

    protected Timer t;

    public DyePack(Color color) {
        this.color = color;
        texture = BaseCode.resources.loadImage(getTexture(color));
        shouldTravel = false;
        visible = false;
        t = new Timer();
    }

    public void initialize(Vector2 center) {
        this.center = center;
        velocity = new Vector2(-Configuration.dyePackSpeed, 0f);

        size.set(width, height);
        shouldTravel = true;
        visible = true;
    }

    public void activate(Hero hero) {
        hero.setColor(color);
        visible = false;
    }

    public static String getTexture(Color color) {
        if (color == Colors.Green) {
            return "Textures/Dye_Green.png";
        }
        if (color == Colors.Blue) {
            return "Textures/Dye_Blue.png";
        }
        if (color == Colors.Yellow) {
            return "Textures/Dye_Yellow.png";
        }
        if (color == Colors.Teal) {
            return "Textures/Dye_Teal.png";
        }
        if (color == Colors.Pink) {
            return "Textures/Dye_Pink.png";
        }
        if (color == Colors.Red) {
            return "Textures/Dye_Red.png";
        }
        return "";
    }

    @Override
    public void update() {
        super.update();
        rotate += 60f * t.deltaTime();
    }

    @Override
    public void handleCollision(Collidable other) {
        if (other instanceof Hero) {
            Hero hero = (Hero) other;
            hero.collect(this);
            DyeHardSound.play(DyeHardSound.pickUpSound);
        }
    }
}

File: DyeHard/CSharp/DyeHard/DyePack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using XNACS1Lib;
using Microsoft.Xna.Framework;

namespace Dyehard
{
    class DyePack : GameObject
    {
        private const float height = 3.5f;
        protected Hero hero;
        protected XNACS1Rectangle box;

        public DyePack(Hero hero, float minX, float maxX, Color color)
        {
            this.hero = hero;

            float padding = hero.getPosition().Width * 2;

            float randomX = XNACS1Base.RandomFloat(minX + padding, maxX - padding);
            float randomY = XNACS1Base.RandomFloat(GameWorld.bottomEdge + padding, GameWorld.topEdge - padding);
            box = new XNACS1Rectangle(new Vector2(randomX, randomY), 0.865f * height, height);
            box.Color = color;
            box.Texture = getTexture(color);
        }

        public override void remove()
        {
            box.RemoveFromAutoDrawSet();
        }

        public void move()
        {
            box.CenterX -= GameWorld.Speed;
        }

        public override void update()
        {
            if (box.Collided(hero.getPosition()) && box.Visible)
            {
                hero.collect(this);
            }
        }

        public void activate()
        {
            hero.setColor(box.Color);
            box.Visible = false;
        }

        public override void draw()
        {
            box.TopOfAutoDrawSet();
        }

        private static String getTexture(Color color)
        {
            if (color == Game.Green) return "Dye_Green";
            if (color == Game.Blue) return "Dye_Blue";
            if (color == Game.Yellow) return "Dye_Yellow";
            if (color == Game.Teal) return "Dye_Teal";
            if (color == Game.Pink) return "Dye_Pink";
            if (color == Game.Red) return "Dye_Red";
            return "";
        }
    }
}
